packages/ui5-proxy-middleware/src/base/utils.ts

```diff
@@ -272,13 +272,13 @@ export const injectScripts = async (
     try {
         const projectRoot = process.cwd();
         const args = process.argv;
-        const htmlFileName = getHtmlFile(req.baseUrl);
+        const htmlFileName = getHtmlFile(req.url);
         const yamlFileName = getYamlFile(args);
         const ui5YamlPath = join(projectRoot, yamlFileName);
         const webAppFolder = await getWebAppFolderFromYaml(ui5YamlPath);
-        const htmlFilePath = join(projectRoot, webAppFolder, htmlFileName);
-        const html = injectUI5Url(htmlFilePath, ui5Configs);
-
+        let htmlFilePath = join(projectRoot, webAppFolder, htmlFileName);
+        htmlFilePath = realpathSync(resolve(projectRoot, htmlFilePath));
```

Comment:
could you please make htmlFilePath a const and put this in one line. It is odd to calculate it and the use the same variable again for a 2nd calculation. So, please const and one line or two lines but with two different variables.

Reply:
btw, are you sure that works? You add the`projectRoot in the join and then again in the resolve. To be honest, the whole thing looks a little odd. Could you describe what the problem is and how you solve it?

Reply:
Done db1f62760fa73d569722c3bdb376383a43a867bc

Reply:
The problem was detected by the code scanning check: https://github.com/SAP/open-ux-tools/security/code-scanning/175 Followed the proposed solution to validate the file path